base/statistics.jl

```diff
@@ -0,0 +1,438 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+##### mean #####
+
+"""
+    mean(itr)
+
+Compute the mean of all elements in a collection.
+
+!!! note
+    If `itr` contains `NaN` or [`missing`](@ref) values, the result is also
+    `NaN` or `missing` (`missing` takes precedence if array contains both).
+    Use the [`skipmissing`](@ref) function to omit `missing` entries and compute the
+    mean of non-missing values.
+
+!!! compat "Julia 1.9"
+    This method requires at least Julia 1.9.
+    Support for older releases is provided by the Statistics package.
+
+# Examples
+```jldoctest
+julia> mean(1:20)
+10.5
+
+julia> mean([1, missing, 3])
+missing
+
+julia> mean(skipmissing([1, missing, 3]))
+2.0
+```
+"""
+mean(itr) = mean(identity, itr)
+
+"""
+    mean(f, itr)
+
+Apply the function `f` to each element of collection `itr` and take the mean.
+
+!!! compat "Julia 1.9"
+    This method requires at least Julia 1.9.
+    Support for older releases is provided by the Statistics package.
+
+# Examples
+```jldoctest
+julia> mean(√, [1, 2, 3])
+1.3820881233139908
+
+julia> mean([√1, √2, √3])
+1.3820881233139908
+```
+"""
+function mean(f, itr)
+    y = iterate(itr)
+    if y === nothing
+        return Base.mapreduce_empty_iter(f, +, itr,
+                                         Base.IteratorEltype(itr)) / 0
```

Comment:
```suggestion
        return mapreduce_empty_iter(f, +, itr,
                                    IteratorEltype(itr)) / 0
```
The `Base.` qualifications are no longer necessary since this is inside of Base. (The same applies to a number of other instances in this file.)